stingray/base.py

```diff
@@ -2134,6 +2134,147 @@ def sort(self, reverse=False, inplace=False):
+        seed : int, default None
+            Random seed to use for the simulation. If None, a random seed is generated.
+
+        """
+        from .gti import get_btis
+        from .utils import get_random_state
+
+        rs = get_random_state(seed)
+
+        if attrs_to_randomize is None:
+            attrs_to_randomize = self.array_attrs() + self.internal_array_attrs()
+            for attr in ["time", "_mask"]:
+                if attr in attrs_to_randomize:
+                    attrs_to_randomize.remove(attr)
+
+        attrs_to_leave_alone = [
+            a
+            for a in self.array_attrs() + self.internal_array_attrs()
+            if a not in attrs_to_randomize
+        ]
+
+        if max_length is None:
+            max_length = np.max(self.gti[:, 1] - self.gti[:, 0]) / 100
+
+        btis = get_btis(self.gti, self.time[0], self.time[-1])
+        if len(btis) == 0:
+            logging.info("No bad time intervals to fill")
+            return copy.deepcopy(self)
+        filtered_times = self.time[self.mask]
+        from .utils import find_nearest
+
+        new_times = [filtered_times.copy()]
+        new_attrs = {}
+        if uniform is None:
+            # The time series is considered uniformly sampled if the median separation between
+            # subsequent times is within 1% of the time resolution
+            uniform = False
+            if self.dt > 0 and np.isclose(np.median(np.diff(self.time)), self.dt, rtol=0.01):
+                uniform = True
+            logging.info(f"Data are {'not' if not uniform else ''} uniformly sampled")
+
+        added_gtis = []
+
+        total_filled_time = 0
+        for bti in btis:
+            length = bti[1] - bti[0]
+            if length > max_length:
+                continue
+            logging.info(f"Filling bad time interval {bti} ({length:.4f} s)")
+            epsilon = 1e-5 * length
+            added_gtis.append([bti[0] - epsilon, bti[1] + epsilon])
+            filt_low_t, filt_low_idx = find_nearest(filtered_times, bti[0])
+            filt_hig_t, filt_hig_idx = find_nearest(filtered_times, bti[1], side="right")
+            if uniform:
+                local_new_times = np.arange(bti[0] + self.dt / 2, bti[1], self.dt)
+                nevents = local_new_times.size
+                new_times.append(local_new_times)
+            else:
+                low_time_arr = filtered_times[max(filt_low_idx - buffer_size, 0) : filt_low_idx]
+                high_time_arr = filtered_times[filt_hig_idx : buffer_size + filt_hig_idx]
+
+                ctrate = (
+                    np.count_nonzero(low_time_arr) / (filt_low_t - low_time_arr[0])
+                    + np.count_nonzero(high_time_arr) / (high_time_arr[-1] - filt_hig_t)
+                ) / 2
+                nevents = rs.poisson(ctrate * (bti[1] - bti[0]))
+                local_new_times = rs.uniform(bti[0], bti[1], nevents)
+                new_times.append(local_new_times)
```

Comment:
Minor thing: This line appears in both if and else statement, so could be pulled out to avoid repetition

Reply:
Fixed